packages/design-system-react/src/components/temp-components/Maskicon/Maskicon.tsx

```diff
@@ -0,0 +1,41 @@
+import DOMPurify from 'dompurify';
+import React, { useEffect, useState } from 'react';
+
+import { MaskiconProps } from './Maskicon.types';
+import { getMaskiconSVG } from './Maskicon.utilities';
+
+export const Maskicon = ({
+  address,
+  size = 32,
+  style,
+  ...props
+}: MaskiconProps) => {
+  const [svgString, setSvgString] = useState('');
+
+  useEffect(() => {
+    let cancelled = false;
+    (async () => {
+      const newSvg = await getMaskiconSVG(address, size);
+      if (!cancelled) {
+        setSvgString(newSvg);
+      }
+    })();
+    return () => {
+      cancelled = true;
+    };
+  }, [address, size]);
+
+  if (!svgString) {
+    return <div style={{ width: size, height: size, ...style }} />;
+  }
+
+  return (
+    <div
+      style={{ width: size, height: size, ...style }}
+      dangerouslySetInnerHTML={{ __html: DOMPurify.sanitize(svgString) }}
```

Comment:
It would be great if we could render the SVG without the wrapping `div` or using `dangerouslySetInnerHTML` have we considered any alternative methods for rendering the SVG in react?

Reply:
I tried rendering the svg string but it didnt work. as far as I know this is the most direct way of doing it